packages/rum-core/src/performance-monitoring/transaction.js

```diff
@@ -196,6 +198,15 @@ class Transaction extends SpanBase {
   isManaged() {
     return !!this.options.managed
   }
+
+  /*
+    Makes sure that a transaction ended while the page was becoming hidden
+    it's not discarded for that very same reason.
+  */
+  endPageHidden(endTime) {
```

Comment:
I created this function because while I was doing the last exploratory testing session I spotted a bug:

The logic where we detect if the website becomes hidden to end the ongoing transaction had a race condition:

1. transaction.end()
2. update lastHiddenStart

unfortunately, the [code where we check](https://github.com/elastic/apm-agent-rum-js/blob/7cd5f5ecbe02ee4ddc3e648927fa4ac0ce3dbb28/packages/rum-core/src/performance-monitoring/transaction-service.js#L258) if a transaction should be discarded because "the page was hidden during the transaction" doesn't happen in a sync way in transaction.end(), it happens after executing a javascript Promise. 

So, assuming that calling `transaction.end()` in that part of the code was a synchronous process was wrong. It was updating the lastHiddenStart before doing the end transaction check (even if the end was invoked before)

**Fortunately**, this was not affecting the page-load transaction, I only reproduced the described behaviour with the INP transaction and also a few times (not all the time, I guess that depended on who was "winning" the race) with clicks transactions.

Reply:
We discussed this offline and planned to add another event `transaction:skipped` to update the last hidden start time.

Reply:
this is the related commit https://github.com/elastic/apm-agent-rum-js/pull/1462/commits/df7423dd9e46545f683d6a6e6db6f5358282d018